test/extended/operators/olm.go

```diff
@@ -164,17 +164,36 @@ var _ = g.Describe("[sig-operator] an end user use OLM", func() {
 		etcdSub             = filepath.Join(buildPruningBaseDir, "etcd-subscription.yaml")
 	)
 
-	files := []string{operatorGroup, etcdSub}
+	files := []string{etcdSub}
 	g.It("can subscribe to the etcd operator", func() {
 		g.By("Cluster-admin user subscribe the operator resource")
+
+		// configure OperatorGroup before tests
+		configFile, err := oc.AsAdmin().Run("process").Args("--ignore-unknown-parameters=true", "-f", operatorGroup, "-p", "NAME=test-operator", fmt.Sprintf("NAMESPACE=%s", oc.Namespace()), "SOURCENAME=community-operators", "SOURCENAMESPACE=openshift-marketplace").OutputToFile("config.json")
+		o.Expect(err).NotTo(o.HaveOccurred())
+		err = oc.AsAdmin().WithoutNamespace().Run("create").Args("-f", configFile).Execute()
+		o.Expect(err).NotTo(o.HaveOccurred())
+		err = wait.Poll(10*time.Second, operatorWait, func() (bool, error) {
+			output, err := oc.AsAdmin().Run("get").Args("-n", oc.Namespace(), "operatorgroup", "test-operator", "-o=jsonpath={.status.targetNamespaces}").Output()
+			if err != nil {
+				e2e.Logf("Failed to get valid operatorgroup, error:%v", err)
+				return false, err
```

Comment:
you should not abort on this error.  you should continue to retry.  (e.g. what if you get a temporary apiserver error on the get?).

The best thing to do is log the error but return nil so you keep retrying until the timeout.